airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/global_parent_cursor.py

```diff
@@ -0,0 +1,197 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Generator, Iterable, Mapping, Optional, Union
+
+from airbyte_cdk.sources.declarative.incremental.declarative_cursor import DeclarativeCursor
+from airbyte_cdk.sources.declarative.partition_routers.partition_router import PartitionRouter
+from airbyte_cdk.sources.types import Record, StreamSlice, StreamState
+
+
+class GlobalParentCursor(DeclarativeCursor):
+    """
+    The GlobalParentCursor is designed to track the state of parent streams using a single global cursor.
```

Comment:
can you update the description to make the constraints super clear? The main one I'm worried about is the fact that the parent and child cursors must be in sync

Reply:
The dependency between the child and parent cursor is not a requirement for this class. The `GlobalParentCursor` can be used to read the parent stream in full refresh mode while using a global cursor for the child stream. This approach is particularly useful when the number of partitions is too large to store individually. By managing the state globally, we can efficiently handle a large number of partitions without exceeding storage limits.

Reply:
update: I see it's in the docs <3

Reply:
I also added a note on how the state is updated